File: 1293/1293.cpp
class Solution {
public:
    int shortestPathInAGridWithObstacl(TreeNode* root) {
        if(!root) return 0;
        return 1+max(shortestPathInAGridWithObstacl(root->left),shortestPathInAGridWithObstacl(root->right));
    }
};

File: 1293/1293.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var shortestPathInAGridWithObstacl = function(root) {
    if (!root) return 0;
    return 1 + Math.max(shortestPathInAGridWithObstacl(root.left), shortestPathInAGridWithObstacl(root.right));
};